Week-3/exercise3.1.js

```diff
@@ -0,0 +1,31 @@
+function memoize(fn) {
+    const cache = new Map()
+    return function(...args) {
+        const key = args.toString()
```

Comment:
What if there are no args? your key will be undefined. Inculcate the habit of considering edge cases while writing program from next time.